src/output.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Output helpers.
+
+// COLORS:
+// The output colour of a particular object of text should convey some information about what
+// exactly it is. These are not hard and fast rules, but generally speaking:
+// - Blue for general information.
+// - Green for success.
+// - Red for errors and warnings.
+// - Cyan should be used for section titles.
+// - Yellow should represent field names -- i.e. UUIDs, versions, etc.
+//
+// By sticking to this guideline we ensure that the CLI stays consistent.
+
+/// Gets a colorized message for outputting to a particular stream. Do not use colors if the stream
+/// is not a tty.
+#[macro_export]
+macro_rules! colorize {
+    ($color:ident, $stream:ident, $arg:tt) => {{
+        let style = if atty::is(atty::Stream::$stream) {
+            ansi_term::Color::$color.normal()
+        } else {
+            ansi_term::Style::default()
+        };
+        style.paint($arg)
+    }};
+    ($color:ident, $stream:ident, $($arg:tt)*) => {{
+        let style = if atty::is(atty::Stream::$stream) {
+            ansi_term::Color::$color.normal()
+        } else {
+            ansi_term::Style::default()
+        };
+        style.paint(format!($($arg)*))
+    }};
+}
+
+/// Wrapper around `print!` that accepts a color as its first argument.
+#[macro_export]
+macro_rules! print_colored {
+    ($color:ident, $($arg:tt)*) => {
+        print!("{}", colorize!($color, Stdout, $($arg)*))
+    };
+}
+
+/// Wrapper around `println!` that accepts a color as its first argument.
+#[macro_export]
+macro_rules! println_colored {
+    ($color:ident, $($arg:tt)*) => {
+        println!("{}", colorize!($color, Stdout, $($arg)*))
+    };
+}
+
+/// Wrapper around `eprint!` that accepts a color as its first argument.
+#[macro_export]
+macro_rules! eprint_colored {
+    ($color:ident, $($arg:tt)*) => {
+        eprint!("{}", colorize!($color, Stderr, $($arg)*))
+    };
+}
+
+/// Wrapper around `eprintln!` that accepts a color as its first argument.
+#[macro_export]
+macro_rules! eprintln_colored {
+    ($color:ident, $($arg:tt)*) => {
+        eprintln!("{}", colorize!($color, Stderr, $($arg)*))
+    };
+}
+
+/// Write an info message to stdout.
+#[macro_export]
+macro_rules! info {
```

Comment:
Is the verbosity level argument coming into play for the `info`, `warn` and `err` macros? Would it be possible to use a logger like `env_var` directly? I think it also handles colors.